### PR TITLE
landscape: allow setting shortcut for leap

### PR DESCRIPTION
**Pull request by matildepark**
- Adds fallback for Leap shortcut in initial state (`meta+/`)
- Moves Leap from the App.js component to the other shortcut contexts in Content
- Allows the Leap shortcut to be set in the settings menu
- Shows the current shortcut on the Leap button

### CODE REVIEW

File: pkg/interface/src/views/components/StatusBar.tsx [hunk 1]
```diff
@@ -36,7 +36,9 @@ const StatusBar = (props) => {
   const invites = [].concat(
     ...Object.values(inviteState).map(obj => Object.values(obj))
   );
-  const metaKey = window.navigator.platform.includes('Mac') ? '⌘' : 'Ctrl+';
+  const leapKey: string = useSettingsState(
+    useCallback(s => s.keyboard['leap'], ['leap'])
+    )?.replace('meta+', '⌘') || '';
```
Comment (arthyn) [inline pkg/interface/src/views/components/StatusBar.tsx:41]: Did you want to always show `⌘` here?

Comment (matildepark) [reply pkg/interface/src/views/components/StatusBar.tsx:41]: If it's a meta shortcut, probably. Do you think it should be agnostic for Linux or the like?

Comment (arthyn) [reply pkg/interface/src/views/components/StatusBar.tsx:41]: Yeah I think so, it seems like y'all were using `window.navigator.platform.includes('Mac') ? '⌘' : 'Ctrl+'` before.

Comment (matildepark) [reply pkg/interface/src/views/components/StatusBar.tsx:41]: Yes, but there's no Linux analogue for the Meta key. Previously, we bound both Ctrl and Meta, and showed Ctrl for other platforms. If you set your key to Meta, what would you refer to it as in a Linux context? Just `Super+`?

Comment (arthyn) [reply pkg/interface/src/views/components/StatusBar.tsx:41]: Oh I see, yeah I guess so, `⌘`, `Win` or `Super`

File: pkg/interface/src/views/landscape/components/Content.tsx [hunk 2]
```diff
@@ -37,6 +37,12 @@ export const Content = (props) => {
     history.goBack();
   }, [history.goBack]));
 
+  useShortcut('leap', useCallback((e) => {
+    e.preventDefault();
+    e.stopImmediatePropagation();
+    props.toggleOmnibox();
+  }, []));
```
Comment (liam-fitzgerald) [inline pkg/interface/src/views/landscape/components/Content.tsx:44]: Needs props.toggleomnibox as a dependency

File: pkg/interface/src/views/components/StatusBar.tsx [hunk 3]
```diff
@@ -36,7 +36,17 @@ const StatusBar = (props) => {
...
+  let metaKey = 'Super+';
+  if (window.navigator.platform.includes('Mac')) {
+    metaKey = '⌘';
+  } else if (window.navigator.platform.includes('Win')) {
+    metaKey = '⊞ Win+';
+  }
+
+  const leapKey: string = useSettingsState(
+    useCallback(s => s.keyboard['leap'], ['leap'])
+    )?.replace('meta+', metaKey) || '';
```
Comment (liam-fitzgerald) [inline pkg/interface/src/views/components/StatusBar.tsx:49]: Most of this should be defined outside the scope of the component and then just pass the function to useSettingsState

**Review by liam-fitzgerald — CHANGES_REQUESTED** [review]
Some nits, else LGTM

File: pkg/interface/src/views/components/StatusBar.tsx (hunk 3 above)
Comment (matildepark) [reply pkg/interface/src/views/components/StatusBar.tsx:49]: I never really understood how that worked with CalmEngine, can you give an example?

Comment (liam-fitzgerald) [reply pkg/interface/src/views/components/StatusBar.tsx:49]:
```typescript
// outside component
  let metaKey = 'Super+';
  if (window.navigator.platform.includes('Mac')) {
    metaKey = '⌘';
  } else if (window.navigator.platform.includes('Win')) {
    metaKey = '⊞ Win+';
  }
const selKey = (s: SettingsState) => s.keyboard.leap.replace('meta+', metaKey);

// inside component

const leapKey = useSettingsState(selKey);
```

**Comment by matildepark** [comment]
All nits addressed, good to stamp and merge

**Review by arthyn — APPROVED** [review]

**Comment by matildepark** [comment]
Going through my dashboard and closing stale PRs ...

**Pull request closed without merging**